### PR TITLE
add vote component

### PR DESCRIPTION
**Pull request by maxime-carabina**
We are adding compoent `Vote`.

image

### CODE REVIEW

**Review by pivilartisant — APPROVED** [review]

**Review by Pierregvx — APPROVED** [review]

File: src/components/Vote/Vote.test.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      onVoteDown: () => null,
+    };
+
+    render(<Vote {...args} />);
+
+    let voteUpNotFilled = screen.getByTestId('up-not-filled');
+    let voteDownNotFilled = screen.getByTestId('down-not-filled');
+
+    expect(voteUpNotFilled).toBeTruthy();
+    expect(voteDownNotFilled).toBeTruthy();
```
Comment (mazmassa) [inline src/components/Vote/Vote.test.tsx:33]:
```suggestion
    expect(voteUpNotFilled).toBeInTheDocument();
    expect(voteDownNotFilled).toBeInTheDocument();
```

File: src/components/Vote/Vote.test.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    let voteUp = screen.getByTestId('vote-up');
+
+    fireEvent.click(voteUp);
+
+    let voteUpFilled = screen.getByTestId('up-filled');
+    let voteDownNotFilled = screen.getByTestId('down-not-filled');
+
+    expect(voteUpFilled).toBeTruthy();
+    expect(voteDownNotFilled).toBeTruthy();
```
Comment (mazmassa) [inline src/components/Vote/Vote.test.tsx:54]:
```suggestion
    expect(voteUpFilled).toBeInTheDocument();
    expect(voteDownNotFilled).toBeInTheDocument();
```

File: src/components/Vote/Vote.test.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    let voteDown = screen.getByTestId('vote-down');
+
+    fireEvent.click(voteDown);
+
+    let voteDownFilled = screen.getByTestId('down-filled');
+    let voteUpNotFilled = screen.getByTestId('up-not-filled');
+
+    expect(voteDownFilled).toBeTruthy();
+    expect(voteUpNotFilled).toBeTruthy();
```
Comment (mazmassa) [inline src/components/Vote/Vote.test.tsx:75]:
```suggestion
    expect(voteDownFilled).toBeInTheDocument();
    expect(voteUpNotFilled).toBeInTheDocument();
```

File: src/components/Vote/Vote.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  customClass?: string;
+}
+
+export function Vote(props: VoteProps) {
+  const { votes, voted, onVoteUp, onVoteDown, customClass, ...rest } = props;
+
+  function onClickItem(arg: 1 | -1) {
+    switch (voted) {
+      case 'VOTE': {
+        arg === 1 ? onVoteDown() : (onVoteDown(), onVoteDown());
```
Comment (mazmassa) [inline src/components/Vote/Vote.tsx:23]: 🤔 do we need the function twice!?

Comment (maxime-carabina) [reply src/components/Vote/Vote.tsx:23]: > thinking do we need the function twice!?

I will simplify the logic...
As it's the backend that has to say whether the user voted `up` or `down`. I'll just replace the 2 voting functions (`onVoteUp` & `onVoteDown`) with `onVote`.

```typescript
<div
  data-testid="vote-up"
  className="cursor-pointer"
  onClick={onVote}
>
  {voted === 'VOTE' ? (
    <TbTriangleFilled data-testid="up-filled" />
  ) : (
    <TbTriangle data-testid="up-not-filled" />
  )}
</div>
```

Comment (mazmassa) [reply src/components/Vote/Vote.tsx:23]: Ohhh much better and beautiful! Great @maxime-carabina !!! 👏

File: src/components/Vote/Vote.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      data-testid="vote"
+      className={`text-f-primary cursor-default ${customClass}`}
+      {...rest}
+    >
+      <div
+        data-testid="vote-up"
+        className="cursor-pointer"
+        onClick={() => onClickItem(1)}
+      >
+        {voted === 'VOTE' ? (
```
Comment (mazmassa) [inline src/components/Vote/Vote.tsx:48]: BTW, extract that for a upper const:

`const isVote = voted === "VOTE"`

then just use here isVote

**Review by mazmassa — APPROVED** [review]

**Comment by mazmassa** [comment]
LGTM! Tks for that! 👏

**Pull request merged**